File: android/src/main/kotlin/com/solusibejo/analytics_debugger/AnalyticsDebuggerPlugin.kt
package com.solusibejo.analytics_debugger

import android.app.Activity
import android.app.Application
import androidx.annotation.NonNull
import app.avo.androidanalyticsdebugger.DebuggerManager
import com.solusibejo.analytics_debugger.consts.MethodNames

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** AnalyticsDebuggerPlugin */
class AnalyticsDebuggerPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel
  private var activity: Activity? = null
  private var debugger: DebuggerManager? = null

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "analytics_debugger")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    if(activity != null && debugger != null){
      when(call.method){
        MethodNames.show -> {
          AnalyticsDebuggerMethods.show(activity!!, debugger!!, call)
        }
        MethodNames.hide -> {
          AnalyticsDebuggerMethods.hide(activity!!, debugger!!)
        }
        MethodNames.send -> {
          AnalyticsDebuggerMethods.send(debugger!!, call)
        }
        else -> result.notImplemented()
      }
    }
    else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }

  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
    activity = binding.activity
    debugger = DebuggerManager(activity!!)
  }

  override fun onDetachedFromActivityForConfigChanges() {
    activity = null
  }

  override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
    activity = binding.activity
    debugger = DebuggerManager(activity!!)
  }

  override fun onDetachedFromActivity() {
    activity = null
  }
}

File: ios/Classes/AnalyticsDebuggerPlugin.swift
import Flutter
import UIKit
import AnalyticsDebugger

public class AnalyticsDebuggerPlugin: NSObject, FlutterPlugin {
    
  let debugger = AnalyticsDebugger()
    
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "analytics_debugger", binaryMessenger: registrar.messenger())
    let instance = AnalyticsDebuggerPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
            case MethodNames.show:
                AnalyticsDebuggerMethods.show(debugger: debugger, call: call)
                break
            case MethodNames.hide:
                AnalyticsDebuggerMethods.hide(debugger: debugger)
                break
            case MethodNames.send:
                AnalyticsDebuggerMethods.send(debugger: debugger, call: call)
                break
            default:
                result(FlutterMethodNotImplemented)
        }
    }
}
